mentat/edit_history.py

```diff
@@ -1,71 +1,88 @@
 import os
 from pathlib import Path
+from typing import Optional
 
 import attr
 from termcolor import colored
 
+from mentat.errors import MentatError
+from mentat.parsers.file_edit import FileEdit, Replacement
+from mentat.session_context import SESSION_CONTEXT
+
 
 # All paths should be abs paths
 @attr.define()
 class RenameAction:
     old_file_name: Path = attr.field()
     cur_file_name: Path = attr.field()
 
-    def undo(self) -> str | None:
+    def undo(self) -> FileEdit:
         if self.old_file_name.exists():
-            return colored(
+            raise MentatError(
```

Comment:
I would prefer to use a new error (rather than MentatError); maybe we could call in UndoError?

Reply:
why not use MentatError? I'm fine with us adding new errors but the main reason in my mind would be if they need to be caught at different places

Reply:
We do catch this error right here in this function; mentaterror is automatically caught outside the main session loop (used to exit with an error message to the user but without a traceback). I would prefer we use a different error to make it clear that we're catching it here in edit_history.